platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/util/PayloadUtils.java

```diff
@@ -16,35 +16,67 @@
 
 package com.swirlds.platform.util;
 
+import static com.swirlds.common.io.streams.SerializableStreamConstants.BOOLEAN_BYTES;
+import static com.swirlds.common.io.streams.SerializableStreamConstants.CLASS_ID_BYTES;
+import static com.swirlds.common.io.streams.SerializableStreamConstants.VERSION_BYTES;
+
+import com.hedera.hapi.platform.event.EventPayload;
 import com.hedera.hapi.platform.event.EventPayload.PayloadOneOfType;
 import com.hedera.hapi.platform.event.StateSignaturePayload;
 import com.hedera.pbj.runtime.OneOf;
 import com.hedera.pbj.runtime.io.buffer.Bytes;
 import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.List;
 
 /**
  * Utility class for handling PJB payloads.
+ * <p>
+ * <b>IMPORTANT:</b> This class is subject to deletion in the future. It's only needed for the transition period
+ * from old serialization to PBJ serialization.
  */
 public final class PayloadUtils {
     private PayloadUtils() {}
 
+    /**
+     * Get the size of an list of payloads.
+     *
+     * @param payloads the payloads to get the size of
+     * @return the size of the payloads
+     */
+    public static int getObjectSize(@NonNull final List<EventPayload> payloads) {
```

Comment:
for both of these methods, I think that their name should indicate that the size check is related to legacy serialization

Reply:
done